src/engine/AmguConsumer.sol

```diff
@@ -19,11 +19,17 @@ abstract contract AmguConsumer is DSMath {
 
     IRegistry public REGISTRY;
 
-    constructor (address _registry) public {
+    constructor(address _registry) public {
         REGISTRY = IRegistry(_registry);
     }
 
     /// @dev if amgu price is zero, skip price fetching
+    modifier amguPayable() {
+        uint256 preGas = gasleft();
+        _;
+        __handleAmguWithIncentiveAndPreGas(0, preGas);
```

Comment:
I think something like this is cleaner:

```
function __chargeAmgu(uint256 gas) private returns (uint256) {
    // do stuff to charge amgu
    return ethCharged;
}

function __chargeIncentive(uint256 amount) private returns (uint256) {
    // do stuff to charge incentive
    return ethCharged;
}

modifier amguPayable() {
    uint256 preGas = gasleft();
    _;
    uint256 postGas = gasleft();
    uint256 ethChargedForAmgu = __chargeAmgu(sub(preGas, postGas));
    anotherHelperToReturnUnspentMsgValue(ethChargedForAmgu);
}

modifier amguPayableWithIncentive() {
    uint256 incentiveAmount = // get from Registry
    uint256 preGas = gasleft();
    _;
    uint256 postGas = gasleft();
    uint256 ethChargedForAmgu = __chargeAmgu(sub(preGas, postGas));
    uint256 ethChargedForIncentive = __chargeIncentive(incentiveAmount);
    anotherHelperToReturnUnspentMsgValue(add(ethChargedForAmgu, ethChargedForIncentive));
}
```

Reply:
Made a change with a modified version of this. Doesn't make sense to have a `chargeIncentive` function imo since there is nothing really happening in there. The incentive is just being sent as part of this tx and there is no call or calculation like in `chargeAmgu`